### PR TITLE
Upgrade and fix import issues with envoy_api

### PR DESCRIPTION
**Pull request by ejona86**
Bazel was not updated as part of feab4e54. Also, `(( X++ ))` exits 1 if X is zero. The patch to data-plane-api is being upstreamed in https://github.com/envoyproxy/envoy/pull/34759

CC @larry-safran

### CODE REVIEW

**Review by sergiitk — COMMENTED** [review]
Since we're basing on the PR that removed xds v2 dependencies, should we also remove `//xds:envoy_service_discovery_v2_java_grpc`?

File: repositories.bzl [hunk 1]
```diff
@@ -142,12 +142,14 @@ def grpc_java_repositories(bzlmod = False):
...
+            sha256 = "c4c9c43903e413924b0cb08e9747f3c3a0727ad221a3c446a326db32def18c60",
+            strip_prefix = "data-plane-api-1611a7304794e13efe2d26f8480a2d2473a528c5",
             urls = [
-                "https://storage.googleapis.com/grpc-bazel-mirror/github.com/envoyproxy/data-plane-api/archive/9d6ffa70677c4dbf23f6ed569676206c4e2edff4.tar.gz",
-                "https://github.com/envoyproxy/data-plane-api/archive/9d6ffa70677c4dbf23f6ed569676206c4e2edff4.tar.gz",
+                "https://storage.googleapis.com/grpc-bazel-mirror/github.com/envoyproxy/data-plane-api/archive/1611a7304794e13efe2d26f8480a2d2473a528c5.tar.gz",
+                "https://github.com/envoyproxy/data-plane-api/archive/1611a7304794e13efe2d26f8480a2d2473a528c5.tar.gz",
             ],
+            patch_args = ["-p1"],
+            patches = ["@io_grpc_grpc_java//:buildscripts/data-plane-api-no-envoy.patch"],
```
Comment (sergiitk) [inline repositories.bzl:152]: We should add a comment with the link to PR/issue in the upstream.

**Review by sergiitk — APPROVED** [review]
Noticed there's a bunch of other references to xds v2 in bazel, so I'm OK if we clean it up in a separate PR

```
❯ rg v2 xds/BUILD.bazel
30:        ":envoy_service_discovery_v2_java_grpc",
32:        ":envoy_service_load_stats_v2_java_grpc",
71:        "@envoy_api//envoy/api/v2:pkg",
72:        "@envoy_api//envoy/api/v2/core:pkg",
73:        "@envoy_api//envoy/api/v2/endpoint:pkg",
74:        "@envoy_api//envoy/config/cluster/aggregate/v2alpha:pkg",
78:        "@envoy_api//envoy/config/filter/http/fault/v2:pkg",
79:        "@envoy_api//envoy/config/filter/http/router/v2:pkg",
80:        "@envoy_api//envoy/config/filter/network/http_connection_manager/v2:pkg",
97:        "@envoy_api//envoy/service/discovery/v2:pkg",
99:        "@envoy_api//envoy/service/load_stats/v2:pkg",
108:    name = "envoy_service_discovery_v2_java_grpc",
109:    srcs = ["@envoy_api//envoy/service/discovery/v2:pkg"],
120:    name = "envoy_service_load_stats_v2_java_grpc",
121:    srcs = ["@envoy_api//envoy/service/load_stats/v2:pkg"],
```

File: repositories.bzl (hunk 1 above)
Comment (ejona86) [reply repositories.bzl:152]: I've updated the patch with the full commit fix, now that it is upstream. I think that addresses this concern.

**Pull request merged**